tests/functional/src/main/java/org/mule/tck/junit4/FunctionalTestCase.java

```diff
@@ -64,26 +66,49 @@ protected String getConfigResources()
         return null;
     }
 
-
     @Override
     protected ConfigurationBuilder getBuilder() throws Exception
     {
+        String domainConfig = getDomainConfig();
+        domainContext = domainConfig != null ? new DomainContextBuilder().setDomainConfig(domainConfig).build() : null;
+        SpringXmlConfigurationBuilder configurationBuilder;
         String configResources = getConfigResources();
         if (configResources != null)
         {
-            return new SpringXmlConfigurationBuilder(configResources);
+            configurationBuilder = new SpringXmlConfigurationBuilder(configResources);
         }
-        configResources = getConfigFile();
-        if (configResources != null)
+        else
         {
-            if (configResources.contains(","))
+            configResources = getConfigFile();
+            if (configResources != null)
+            {
+                if (configResources.contains(","))
+                {
+                    throw new RuntimeException("Do not use this method when the config is composed of several files. Use getConfigFiles method instead.");
+                }
+                configurationBuilder = new SpringXmlConfigurationBuilder(configResources);
+            }
+            else
             {
-                throw new RuntimeException("Do not use this method when the config is composed of several files. Use getConfigFiles method instead.");
+                String[] multipleConfigResources = getConfigFiles();
+                configurationBuilder = new SpringXmlConfigurationBuilder(multipleConfigResources);
             }
-            return new SpringXmlConfigurationBuilder(configResources);
         }
-        String[] multipleConfigResources = getConfigFiles();
-        return new SpringXmlConfigurationBuilder(multipleConfigResources);
+        if (domainContext != null)
+        {
+            configurationBuilder.setDomainContext(domainContext);
+        }
+        return configurationBuilder;
+    }
+
+    @Override
+    protected void doTearDownAfterMuleContextDispose() throws Exception
+    {
+        super.doTearDownAfterMuleContextDispose();
+        if (domainContext != null)
+        {
+            domainContext.dispose();
```

Comment:
In AbstractMuleContextTestCase muleContext is disposed only if this condition applies:

muleContext != null && !(muleContext.isDisposed() || muleContext.isDisposing()))

I don't know if there is a reason for this, but if there is we should consider it for domainContext.